core/types/transaction.go

```diff
@@ -649,7 +649,15 @@ func NewMessage(from common.Address, to *common.Address, nonce uint64, amount *b
 }
 
 // AsMessage returns the transaction as a core.Message.
-func (tx *Transaction) AsMessage(s Signer, baseFee, l1Cost *big.Int) (Message, error) {
+func (tx *Transaction) AsMessage(s Signer, fees ...*big.Int) (Message, error) {
```

Comment:
Can we instead use an Option pattern? This way not each optional argument has to be a `big.Int`, and we can add more in the future.

E.g.
```go
type MsgOption interface {
  Apply(tx *Transaction, msg *Message) error
}

type MsgOptionFn func(tx *Transaction, msg *Message) error

func (fn MsgOptionFn) Apply(tx *Transaction, msg *Message) error {
  return fn(tx, msg)
}

func (tx *Transaction) AsMessage(s Signer, opts ...MsgOption) (Message, error) {
   msg := Message{...}
   for i, opt := range opts {
     if err := opt.Apply(tx, msg); err != nil {
       return nil, fmt.Errorf("failed to apply option %d: %w", i, err)
     }
   }
  ...
}
...


func L1CostOption(cost *big.Int) MsgOption {
  return MsgOptionFn(func(tx *Transaction, msg *Message) error {
    msg.l1Cost = cost
    return nil
  }
}
```

Reply:
This works, I opted to keep `baseFee` outside the Options as a normal func param though, as it simplifies the diff since we don't need to modify callers.

Reply:
Makes sense for `baseFee`, it's not optional, and already in every existing caller, so we shouldn't change that. If upstream ever adopts the option pattern then we might change it.